shell/platform/windows/windows_lifecycle_manager.h

```diff
@@ -34,12 +47,43 @@ class WindowsLifecycleManager {
                     std::optional<LPARAM> lparam,
                     UINT exit_code);
 
-  // Intercept top level window messages, only paying attention to WM_CLOSE.
+  // Intercept top level window WM_CLOSE message and listen to events that may
+  // update the application lifecycle.
   bool WindowProc(HWND hwnd, UINT msg, WPARAM w, LPARAM l, LRESULT* result);
 
   // Signal to start consuming WM_CLOSE messages.
   void BeginProcessingClose();
 
+  // Update the app lifecycle state in response to a change in window state.
+  // When the app lifecycle state actually changes, this sends a platform
+  // message to the framework notifying it of the state change.
+  virtual void SetLifecycleState(AppLifecycleState state);
+
+  // Respond to a change in window state. Transitions as follows:
+  // When the only visible window is hidden, transition from resumed or
+  // inactive to hidden.
+  // When the only focused window is unfocused, transition from resumed to
+  // inactive.
+  // When a window is focused, transition from inactive to resumed.
+  // When a window is shown, transition from hidden to inactive.
+  virtual void OnWindowStateEvent(HWND hwnd, WindowStateEvent event);
+
+  AppLifecycleState GetLifecycleState() { return state_; }
```

Comment:
Nit:

```suggestion
  AppLifecycleState lifecycle_state() { return state_; }
```

Also, consider moving this to be the last public member. I don't feel strongly, feel free to skip.